### PR TITLE
AST: clean up AST tests

### PR DESCRIPTION
**Pull request by helixbass**
@GeoffreyBooth per your suggestion, PR to go through AST tests and clean up remaining commented-out tests

Based on `ast-test-compile-errors`, [here](https://github.com/helixbass/copheescript/compare/ast-test-compile-errors...ast-clean-up-tests) is just the diff against that branch

### CODE REVIEW

File: src/grammar.coffee [hunk 1]
```diff
@@ -493,12 +493,12 @@ grammar =
...
     o 'EXPORT { ExportSpecifierList OptComma }',             -> new ExportNamedDeclaration new ExportSpecifierList $3
     o 'EXPORT Class',                                        -> new ExportNamedDeclaration $2
-    o 'EXPORT Identifier = Expression',                      -> new ExportNamedDeclaration new Assign $2, $4, null,
-                                                                                                      moduleDeclaration: 'export'
-    o 'EXPORT Identifier = TERMINATOR Expression',           -> new ExportNamedDeclaration new Assign $2, $5, null,
-                                                                                                      moduleDeclaration: 'export'
-    o 'EXPORT Identifier = INDENT Expression OUTDENT',       -> new ExportNamedDeclaration new Assign $2, $5, null,
-                                                                                                      moduleDeclaration: 'export'
+    o 'EXPORT Identifier = Expression',                      -> new ExportNamedDeclaration LOC(2,4)(new Assign $2, $4, null,
```
Comment (helixbass) [inline src/grammar.coffee]: Inaccurate location data was being generated for eg `export fn = ->`

File: test/abstract_syntax_tree.coffee [hunk 2]
```diff
@@ -769,25 +782,6 @@ test "AST as expected for AwaitReturn node", ->
             argument: NUMBER 2
       ]
 
-# test "AST as expected for Value node", ->
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: `Value` doesn't need its own section of AST tests, the different types of `Value`'s are tested elsewhere (eg basic values with properties are tested under `Access`)

File: test/abstract_syntax_tree.coffee [hunk 3]
```diff
@@ -1329,28 +1323,6 @@ test "AST as expected for Range node", ->
     to:
       value: 2
 
-  # testExpression 'for x in [42...43] then',
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: `for` + range is tested under `For`

File: test/abstract_syntax_tree.coffee [hunk 4]
```diff
@@ -1562,10 +1534,6 @@ test "AST as expected for Obj node", ->
     ]
     implicit: yes
 
-#   # TODO: Test destructuring.
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: Object destructuring is tested elsewhere eg under `Assign`

File: test/abstract_syntax_tree.coffee [hunk 5]
```diff
@@ -1579,8 +1547,6 @@ test "AST as expected for Arr node", ->
       {operator: '!'}
     ]
 
-#   # TODO: Test destructuring.
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: Array destructuring is tested elsewhere eg under `Assign`

File: test/abstract_syntax_tree.coffee [hunk 6]
```diff
@@ -2691,14 +2671,15 @@ test "AST as expected for Splat node", ->
       postfix: no
     ]
 
-#   # TODO: Test object splats.
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: Object splats are tested elsewhere eg under `Obj`

File: test/abstract_syntax_tree.coffee [hunk 7]
```diff
@@ -3731,8 +3709,6 @@ test "AST as expected for Switch node", ->
       ]
     ]
 
-#   # TODO: File issue for compile error when using `then` or `;` where `\n` is rn.
```
Comment (helixbass) [inline test/abstract_syntax_tree.coffee]: I'm not sure what this was referring to

Comment (GeoffreyBooth) [reply test/abstract_syntax_tree.coffee]: Probably `then\r\n` as opposed to the typical `then\n`.

Comment (Inve1951) [reply test/abstract_syntax_tree.coffee]: I forgot that there. It's covered in #5127.

**Comment by GeoffreyBooth** [comment]
✅ once conflicts resolved.

**Comment by helixbass** [comment]
@GeoffreyBooth ok rebased

**Pull request merged**